src/inputs/dropdown/query-search-dropdown/PQuerySearchDropdown.vue

```diff
@@ -0,0 +1,255 @@
+                           :placeholder="querySearchState.currentPlaceholder || scope.placeholder"
+                           :type="querySearchState.inputElType"
+                           :step="querySearchState.currentDataType === 'integer' ? 1 : undefined"
+                           :min="querySearchState.currentDataType === 'integer' ? 0 : undefined"
+                           @input="onInput"
+                           @keyup.enter="onEnter"
+                           @keydown="onKeydownCheck"
+                           @click.stop="showMenu(true)"
+                           @focus="focus"
+                           @blur="blur"
+                           @paste="onPaste"
+                    >
+                </span>
+            </template>
+            <template #right="scope">
+                <div class="right">
+                    <span v-if="querySearchState.selectedKey || scope.value" class="delete-button" @click="onDeleteAll">
+                        <p-i class="icon" name="ic_delete" height="1rem"
+                             width="1rem"
+                        />
+                    </span>
+                    <span class="dropdown-button" :class="{'text-blue-600': querySearchState.isFocused}" @click="handleClickDropdownButton">
+                        <p-i class="icon" :name="querySearchState.visibleMenu ? 'ic_arrow_top' : 'ic_arrow_bottom'"
+                             color="inherit"
+                        />
+                    </span>
+                </div>
+            </template>
+        </p-search>
+        <div v-show="querySearchState.visibleMenu" class="menu-container">
+            <p-context-menu ref="menuRef"
+                            :loading="querySearchState.lazyLoading"
+                            :menu="querySearchState.menu"
+                            :highlight-term="querySearchState.searchText"
+                            no-select-indication
+                            @keyup:up:end="focus"
+                            @keyup:down:end="focus"
+                            @select="onMenuSelect"
+                            @blur="focus"
+            />
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import type { PropType, DirectiveFunction, SetupContext } from 'vue';
+import {
+    defineComponent, reactive, toRefs,
+} from 'vue';
+
+import { vOnClickOutside } from '@vueuse/components';
+import { focus as vFocus } from 'vue-focus';
+
+import PI from '@/foundation/icons/PI.vue';
+import { useProxyValue } from '@/hooks';
+import { useQuerySearch } from '@/hooks/query-search';
+import { PTag } from '@/index';
+import PContextMenu from '@/inputs/context-menu/PContextMenu.vue';
+import type { SearchDropdownMenuItem } from '@/inputs/dropdown/search-dropdown/type';
+import type { KeyMenuItem, ValueMenuItem, QueryItem } from '@/inputs/search/query-search/type';
+import PSearch from '@/inputs/search/search/PSearch.vue';
+
+export default defineComponent({
```

Comment:
Please set props type.

```suggestion
export default defineComponent<QuerySearchDropdownProps>({
```